### PR TITLE
Add spring boot starter documentation

### PR DESCRIPTION
**Pull request by gurbuzali**
Add description here

Checklist
- [X] Tags Set
- [X] Milestone Set
- [ ] Any breaking changes are documented
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] For code samples, code sample main readme is updated

Links to issues fixed (if any):

Fixes #NNNN

List of breaking changes:

* ..

### CODE REVIEW

File: site/docs/api/spring.md [hunk 1]
```diff
@@ -328,3 +328,23 @@ attributes.
     </jet:client>
 </beans>
 ```
+
+## SpringBoot Integration
```
Comment (eminn) [inline site/docs/api/spring.md:332]: space needed here ?

File: site/docs/api/spring.md [hunk 2]
```diff
@@ -328,3 +328,23 @@ attributes.
...
+
+Hazelcast Jet provides
+[Hazelcast Jet Spring Boot Starter](https://github.com/hazelcast/hazelcast-jet-contrib/tree/master/hazelcast-jet-spring-boot-starter)
+which automatically configures and starts a `JetInstance`, either a
+server or a client, if Jet is on the classpath.
+
```
Comment (cangencer) [inline site/docs/api/spring.md:338]: can we give an example how to use it?

Comment (gurbuzali) [reply site/docs/api/spring.md:338]: I've added a tutorial and linked it here

File: site/docs/tutorials/spring-boot.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+```java
+package org.example;
+
+import com.hazelcast.jet.JetInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (eminn) [inline site/docs/tutorials/spring-boot.md:85]: Do we need this `@RestController` since we are not defining any endpoints ?

Comment (gurbuzali) [reply site/docs/tutorials/spring-boot.md:85]: it's a leftover

File: site/docs/tutorials/spring-boot.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+Let's add some custom configuration to our Jet member by defining a
+configuration file named `hazelcast-jet.yaml` at the root directory.
+
+```yaml
+hazelcast-jet:
+  instance:
+    cooperative-thread-count: 4
+  edge-defaults:
+    queue-size: 2048
+  hazelcast:
```
Comment (eminn) [inline site/docs/tutorials/spring-boot.md:118]: This doesn't seem like a correct yaml configuration

Comment (gurbuzali) [reply site/docs/tutorials/spring-boot.md:118]: weird, it worked actually while writing the tutorial, I've observed that the there was a log `cooperative-thread-count= 4`

Comment (eminn) [reply site/docs/tutorials/spring-boot.md:118]: I mean the `hazelcast` part

**Pull request merged**